File: blog-api/src/main/scala/su/wps/blog/models/domain/PostTranslation.scala
package su.wps.blog.models.domain

import io.circe.Encoder

import java.time.ZonedDateTime

final case class PostTranslation(
  postId: PostId,
  languageCode: String,
  name: String,
  text: Option[String],
  shortText: Option[String],
  seoTitle: Option[String],
  seoDescription: Option[String],
  seoKeywords: Option[String],
  translationStatus: String,
  createdAt: ZonedDateTime,
  updatedAt: ZonedDateTime,
  id: Option[PostTranslationId] = None
)

final case class PostTranslationId(value: Int) extends AnyVal

object PostTranslationId {
  implicit val encoder: Encoder[PostTranslationId] = Encoder[Int].contramap(_.value)
}

File: blog-admin/blog_admin/blog_admin/models/post_translation.py
from django.db import models


class PostTranslation(models.Model):
    TRANSLATION_STATUS_CHOICES = [
        ('draft', 'Draft'),
        ('in_progress', 'In Progress'),
        ('ready', 'Ready'),
        ('published', 'Published'),
    ]

    post = models.ForeignKey(
        'Post', on_delete=models.CASCADE, related_name='translations'
    )
    language = models.ForeignKey(
        'Language',
        on_delete=models.CASCADE,
        db_column='language_code',
        to_field='code',
    )
    name = models.TextField()
    text = models.TextField(blank=True, null=True)
    short_text = models.TextField(blank=True, null=True)
    seo_title = models.CharField(max_length=255, blank=True, null=True)
    seo_description = models.TextField(blank=True, null=True)
    seo_keywords = models.TextField(blank=True, null=True)
    translation_status = models.CharField(
        max_length=20, choices=TRANSLATION_STATUS_CHOICES, default='draft'
    )
    search_vector = models.TextField(blank=True, null=True)
    created_at = models.DateTimeField(auto_now_add=True)
    updated_at = models.DateTimeField(auto_now=True)

    class Meta:
        managed = False
        db_table = 'post_translations'
        unique_together = [('post', 'language')]

    def __str__(self):
        return f"{self.name} [{self.language_id}]"
